src/Mod/PartDesign/App/FeatureSketchBased.cpp

```diff
@@ -666,6 +705,154 @@ void ProfileBased::generatePrism(TopoDS_Shape& prism,
+    gp_Vec vecOffset = gp_Vec(direction) * Length;
+    gp_Vec vecOffset2 = gp_Vec(direction.Reversed()) * Length2;
+
+    // create a new shape which is the from shape with the offset defined by the TaperAngle
+    // at first inspect the from shape
+    int numEdges = 0;
+    TopExp_Explorer xp(sourceWire, TopAbs_EDGE);
+    while (xp.More()) {
+        numEdges++;
+        xp.Next();
+    }
+    // Pad / Pocket does not only support closed wires, thus only 1 edge is not valid
+    // if this might be supported in future, see Extrusion::makeDraft how to handle 1 edge
+    if (numEdges == 1)
+        throw Base::RuntimeError("Base shape consists of only one edge");
+
+    // to store the sections for the loft
+    std::list<TopoDS_Wire> list_of_sections;
+    BRepBuilderAPI_MakeWire sectionResult;
+    
+    // if we have a second direction length create a bottom section
+    if (Length2 > Precision::Confusion()) {
+        // this offsetWire must be the first loft section
+        createTaperedPrismOffset(sourceWire, vecOffset2, lateralOffset2, Length2, true, sectionResult);
+        list_of_sections.push_back(sectionResult);
+    }
+
+    // since the order is important store the other sections after bottom
+    list_of_sections.push_back(sourceWire);
+    // create the top section
+    createTaperedPrismOffset(sourceWire, vecOffset, lateralOffset, Length, false, sectionResult);
+    list_of_sections.push_back(sectionResult);
+    
+    // build the loft using the created sections
+    BRepOffsetAPI_ThruSections mkGenerator(Standard_True, /*ruled=*/Standard_True);
+    for (std::list<TopoDS_Wire>::const_iterator it = list_of_sections.begin(); it != list_of_sections.end(); ++it) {
+        const TopoDS_Wire& wire = *it;
+        mkGenerator.AddWire(wire);
+    }
+
+    try {
+#if defined(__GNUC__) && defined (FC_OS_LINUX)
+        Base::SignalException se;
+#endif
+        mkGenerator.Build();
+        drafts.push_back(mkGenerator.Shape());
+    }
+    catch (Standard_Failure&) {
+        throw;
+    }
+    catch (...) {
+        throw Base::CADKernelError("Unknown exception from BRepOffsetAPI_ThruSections");
+    }
+}
+
+void ProfileBased::createTaperedPrismOffset(TopoDS_Wire sourceWire,
+                                            const gp_Vec& translation,
+                                            double offset,
+                                            const double Length,
+                                            bool isSecond,
+                                            BRepBuilderAPI_MakeWire& result) {
+    gp_Trsf tempTransform;
+    tempTransform.SetTranslation(translation);
+    TopLoc_Location loc(tempTransform);
+    TopoDS_Wire movedSourceWire = TopoDS::Wire(sourceWire.Moved(loc));
+
+    TopoDS_Shape offsetShape;
+    if (fabs(offset) > Precision::Confusion()) {
+        TopLoc_Location wireLocation;
+        TopLoc_Location edgeLocation;
+        BRepOffsetAPI_MakeOffset mkOffset;

```

Comment:
Without precompiled headers, this requires an `# include <BRepOffsetAPI_MakeOffset.hxx>` here on apple-clang-11.0

Reply:
thanks, added